bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java

```diff
@@ -0,0 +1,306 @@
+                break;
+            case "Setpoint":
+                SetpointImpl setpointWidget = (SetpointImpl) SitemapFactory.eINSTANCE.createSetpoint();
+                widget = setpointWidget;
+                setWidgetPropertyFromComponentConfig(widget, component, "minValue", SitemapPackage.SETPOINT__MIN_VALUE);
+                setWidgetPropertyFromComponentConfig(widget, component, "maxValue", SitemapPackage.SETPOINT__MAX_VALUE);
+                setWidgetPropertyFromComponentConfig(widget, component, "step", SitemapPackage.SETPOINT__STEP);
+                break;
+            case "Colorpicker":
+                ColorpickerImpl colorpickerWidget = (ColorpickerImpl) SitemapFactory.eINSTANCE.createColorpicker();
+                widget = colorpickerWidget;
+                setWidgetPropertyFromComponentConfig(widget, component, "frequency",
+                        SitemapPackage.COLORPICKER__FREQUENCY);
+                break;
+            case "Default":
+                DefaultImpl defaultWidget = (DefaultImpl) SitemapFactory.eINSTANCE.createDefault();
+                widget = defaultWidget;
+                setWidgetPropertyFromComponentConfig(widget, component, "height", SitemapPackage.DEFAULT__HEIGHT);
+                break;
+            default:
+                logger.warn("Unknown sitemap component type {}", component.getType());
+                break;
+        }
+
+        if (widget != null) {
+            setWidgetPropertyFromComponentConfig(widget, component, "label", SitemapPackage.WIDGET__LABEL);
+            setWidgetPropertyFromComponentConfig(widget, component, "icon", SitemapPackage.WIDGET__ICON);
+            setWidgetPropertyFromComponentConfig(widget, component, "item", SitemapPackage.WIDGET__ITEM);
+
+            if (widget instanceof LinkableWidget) {
+                LinkableWidget linkableWidget = (LinkableWidget) widget;
+                if (component.getSlots() != null && component.getSlots().containsKey("widgets")) {
+                    for (UIComponent childComponent : component.getSlot("widgets")) {
+                        Widget childWidget = buildWidget(childComponent);
+                        if (childWidget != null) {
+                            linkableWidget.getChildren().add(childWidget);
+                        }
+                    }
+                }
+            }
+
+            // TODO: process visibility & color rules
+        }
+
+        return widget;
+    }
+
+    private void setWidgetPropertyFromComponentConfig(Widget widget, UIComponent component, String configParamName,
+            int feature) {
+        if (component == null || component.getConfig() == null) {
+            return;
+        }
+        Object value = component.getConfig().get(configParamName);
+        if (value == null) {
+            return;
+        }
+        if (value instanceof Double) {
```

Comment:
If you do a `ConfigUtil.normalizeTypes()` on the config, you won't have to deal with Double values and other exceptions.